src/XProfile/VirtoCommerce.XProfile/Queries/GetContactByIdQueryHandler.cs

```diff
@@ -12,9 +12,9 @@ public GetContactByIdQueryHandler(IContactAggregateRepository contactAggregateRe
         {
             _contactAggregateRepository = contactAggregateRepository;
         }
-        public Task<ContactAggregate> Handle(GetContactByIdQuery request, CancellationToken cancellationToken)
+        public async Task<ContactAggregate> Handle(GetContactByIdQuery request, CancellationToken cancellationToken)
         {
-            return _contactAggregateRepository.GetContactByIdAsync(request.ContactId);
+            return (ContactAggregate) await _contactAggregateRepository.GetMemberAggregateRootByIdAsync(request.ContactId);
```

Comment:
We can remove the async/await combination here if we remove the type conversion.